keanu-python/keanu/vertex/base.py

```diff
@@ -202,6 +211,87 @@ class Integer(Vertex):
+                return result
+            except KeyError:
+                return super().__array_ufunc__(ufunc, method, input0, input1)
+        else:
+            raise NotImplementedError("NumPy ufunc method %s not implemented" % method)
+
+    def __op_based_on_other_type(self, other: vertex_operation_param_types, op: Callable,
+                                 integer_op_ctr: Callable) -> 'Vertex':
+        if is_floating_type(other):
+            # Equivalent to kn.vertex.generated.CastToDouble(self).__add__(other) for add
+            return op(kn.vertex.generated.CastToDouble(self))
+        else:
+            return integer_op_ctr(other)
+
+    def __add__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(
+            other, lambda casted_to_double: casted_to_double.__add__(other),
+            lambda other_holder: kn.vertex.generated.IntegerAddition(self, other_holder))
+
+    def __radd__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(
+            other, lambda casted_to_double: casted_to_double.__radd__(other),
+            lambda other_holder: kn.vertex.generated.IntegerAddition(other_holder, self))
+
+    def __sub__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(
+            other, lambda casted_to_double: casted_to_double.__sub__(other),
+            lambda other_holder: kn.vertex.generated.IntegerDifference(self, other_holder))
+
+    def __rsub__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(
+            other, lambda casted_to_double: casted_to_double.__rsub__(other),
+            lambda other_holder: kn.vertex.generated.IntegerDifference(other_holder, self))
+
+    def __mul__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(
+            other, lambda casted_to_double: casted_to_double.__mul__(other),
+            lambda other_holder: kn.vertex.generated.IntegerMultiplication(self, other_holder))
+
+    def __rmul__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(
+            other, lambda casted_to_double: casted_to_double.__rmul__(other),
+            lambda other_holder: kn.vertex.generated.IntegerMultiplication(other_holder, self))
+
+    def __pow__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(other, lambda casted_to_double: casted_to_double.__pow__(other),
+                                             lambda other_holder: kn.vertex.generated.IntegerPower(self, other_holder))
+
+    def __rpow__(self, other: vertex_operation_param_types) -> 'Vertex':
+        return self.__op_based_on_other_type(other, lambda casted_to_double: casted_to_double.__rpow__(other),
```

Comment:
nit: format the same as the others (CR before `other`)

Reply:
That fails the `yapf` formatting checks